runtime/wasmer.go

```diff
@@ -93,11 +94,23 @@ func NewRuntime(code []byte, t *trie.Trie) (*Runtime, error) {
 	data := unsafe.Pointer(&index)
 	instance.SetContextData(data)
 
-	return &Runtime{
+	r := &Runtime{
 		vm:    instance,
 		trie:  t,
 		mutex: sync.Mutex{},
-	}, nil
+	}
+
+	// Clean up the registry if r is GC'd
+	runtime.SetFinalizer(r, func(_ *Runtime) {
+		// Launch a goroutine to avoid blocking the GC...
+		go func() {
+			mutex.Lock()
+			delete(registry, index)
+			mutex.Unlock()
+		}()
```

Comment:
I think this is already inside its own goroutine so it won't block the GC: https://golang.org/pkg/runtime/#SetFinalizer

Reply:
My documentation that this is to prevent blocking the GC is slightly incorrect. This won't block the GC but it can block other finalizers from running. See the last line of the documentation for `runtime.SetFinalizer`

> A single goroutine runs all finalizers for a program, sequentially. If a finalizer must run for a long time, it should do so by starting a new goroutine.

Reply:
Wicked, thanks!